src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -62,10 +62,11 @@ internal partial class Binder
                 if (source.Kind == BoundKind.TupleLiteral)
                 {
                     var sourceTuple = (BoundTupleLiteral)source;
+                    TupleTypeSymbol.ReportNamesMismatches(destination.TupleElementNames, sourceTuple, diagnostics);
                     source = new BoundConvertedTupleLiteral(
                         sourceTuple.Syntax,
-                        sourceTuple.Type, 
-                        sourceTuple.Arguments, 
+                        sourceTuple.Type,
+                        sourceTuple.Arguments,
                         sourceTuple.Type, // same type to keep original element names 
                         sourceTuple.HasErrors);
```

Comment:
At this point, we're only issuing a warning, so I think the bound nodes are good (no error).